apps/gdalalg_raster_overview_refresh.cpp

```diff
@@ -0,0 +1,561 @@
+/*              GDALRasterOverviewAlgorithmRefresh::RunImpl()           */
+/************************************************************************/
+
+bool GDALRasterOverviewAlgorithmRefresh::RunImpl(GDALProgressFunc pfnProgress,
+                                                 void *pProgressData)
+{
+    auto poDS = m_dataset.GetDatasetRef();
+    CPLAssert(poDS);
+    if (poDS->GetRasterCount() == 0)
+    {
+        ReportError(CE_Failure, CPLE_AppDefined, "Dataset has no raster band");
+        return false;
+    }
+
+    auto poBand = poDS->GetRasterBand(1);
+    const int nOvCount = poBand->GetOverviewCount();
+
+    std::vector<int> levels = m_levels;
+
+    // If no levels are specified, reuse the potentially existing ones.
+    if (levels.empty())
+    {
+        for (int iOvr = 0; iOvr < nOvCount; ++iOvr)
+        {
+            auto poOverview = poBand->GetOverview(iOvr);
+            if (poOverview)
+            {
+                const int nOvFactor = GDALComputeOvFactor(
+                    poOverview->GetXSize(), poBand->GetXSize(),
+                    poOverview->GetYSize(), poBand->GetYSize());
+                levels.push_back(nOvFactor);
+            }
+        }
+    }
+    if (levels.empty())
+    {
+        ReportError(CE_Failure, CPLE_AppDefined, "No overviews to refresh");
+        return false;
+    }
+
+    std::vector<int> anOvrIndices;
+    for (int nLevel : levels)
+    {
+        int nIdx = -1;
+        for (int iOvr = 0; iOvr < nOvCount; iOvr++)
+        {
+            auto poOverview = poBand->GetOverview(iOvr);
+            if (poOverview)
+            {
+                const int nOvFactor = GDALComputeOvFactor(
+                    poOverview->GetXSize(), poBand->GetXSize(),
+                    poOverview->GetYSize(), poBand->GetYSize());
+                if (nOvFactor == nLevel ||
+                    nOvFactor == GDALOvLevelAdjust2(nLevel, poBand->GetXSize(),
+                                                    poBand->GetYSize()))
+                {
+                    nIdx = iOvr;
+                    break;
+                }
+            }
+        }
+        if (nIdx < 0)
+        {
+            CPLError(CE_Failure, CPLE_AppDefined,
+                     "Cannot find overview level with subsampling factor of %d",
+                     nLevel);
+            return false;
+        }
+        CPLDebug("GDAL", "Refreshing overview idx %d", nIdx);
+        anOvrIndices.push_back(nIdx);
+    }
+
+    std::string resampling = m_resampling;
+    if (resampling.empty())
+    {
+        const char *pszResampling =
+            poBand->GetOverview(0)->GetMetadataItem("RESAMPLING");
+        if (pszResampling)
+        {
+            resampling = pszResampling;
+            CPLDebug("GDAL",
+                     "Reusing resampling method %s from existing "
+                     "overview",
+                     pszResampling);
+        }
+    }
+    if (resampling.empty())
+        resampling = "nearest";
```

Comment:
shouldn't this be set as a default in the Arg ?

Reply:
not really, as a default value in GDALAlgorithmArg implies it will be used if unspecified. Here the value got from the RESAMPLING metadata item if found will have precedence